grammar.js

```diff
@@ -215,15 +217,56 @@ module.exports = grammar({
       )
     )),
 
+    _literal: $ => choice(
+      $.symbol,
+      $.integer,
+      $.float,
+      $.boolean,
+      $.nil,
+      $.string
+    ),
+
     symbol: $ => token(seq(':', choice(identifierChars(), operatorChars()))),
+    integer: $ => (/0b[01](_?[01])*|0[oO]?[0-7](_?[0-7])*|(0d)?\d(_?\d)*|0x[0-9a-fA-F](_?[0-9a-fA-F])*/),
+    float: $ => (/\d(_?\d)*\.\d(_?\d)*([eE]\d(_?\d)*)?/),
+    boolean: $ => choice('true', 'false', 'TRUE', 'FALSE'),
+    nil: $ => choice('nil', 'NIL'),
+
+    string: $ => choice(
+      $._single_quoted,
+      $._double_quoted,
+      $._percent_q
+    ),
+    _single_quoted: $ => stringBody("'"),
+    _double_quoted: $ => token(seq('"', repeat(choice(/\\./, $.interpolation, /[^\\"]/)), '"')),
+    _percent_q: $ => choice(
+      token(seq('%q', choice.apply(null, unbalancedDelimiters.split('').map(stringBody)))),
+      seq('%q', $._percent_q_angle),
+      seq('%q', $._percent_q_bracket),
+      seq('%q', $._percent_q_paren),
+      seq('%q', $._percent_q_brace)
```

Comment:
It seems like if I can wrap `_double_quoted` in `token`, then I should be able to do that with these as well. However, if I do that they don’t parse correctly.

Thus, these rules are slightly wrong: they’ll allow whitespace :confused: